src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2109,12 +2109,18 @@ private unsafe ParsingError PrivateParseMinimal()
                             _flags |= Flags.UncPath;
                             idx = i;
                         }
+                        else if (!IsWindowsSystem && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && i - idx == 3 && pUriString[i - 1] == '/')
+                        {
```

Comment:
I propose we add `file://localhost/` to this PR. This means Uri's that start with `/`, `file:///` and `file://localhost/` will preserve backslashes on Unix.
Non localhost uri's remain Unc uris.
What do you think?